src/OrchardCore.Modules/OrchardCore.ContentFields/Indexing/SQL/LinkFieldIndexProvider.cs

```diff
@@ -89,11 +90,11 @@ public override void Describe(DescribeContext<ContentItem> context)
                                 ContentType = contentItem.ContentType,
                                 ContentPart = pair.Definition.ContentTypePartDefinition.Name,
                                 ContentField = pair.Definition.Name,
-                                Url = pair.Field.Url?[..Math.Min(pair.Field.Url.Length, LinkFieldIndex.MaxUrlSize)],
+                                Url = pair.Field.Url?.Substring(0, Math.Min(pair.Field.Url.Length, LinkFieldIndex.MaxUrlSize)),
```

Comment:
Why go back to `Substring()`?

Reply:
While I agree, there are many cases of it in the source code, and just 5 for `Substring()`:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/d851291f-b845-4933-8b74-1c5e792def6f)

(And 3 more for another overload.)

At one point, the new syntax was adopted, so I'd prefer consistency. We can keep this now, but we'll need to have a formal recommendations once we enforce either of these with analyzers.

Reply:
At some point we changed that preference in the `.editorconfig` file so even the IDE no longer suggest to use this syntax. This was discussed one time in a meeting and the suggestion was to use a readable code over the new fancy syntax. I think this suggestion came from @sebastienros